autopilot-daemon/pkg/handlers/healthchecks.go

```diff
@@ -22,11 +22,35 @@ func PeriodicCheckTimer() {
 }
 
 func InvasiveCheckTimer() {
-	klog.Info("Trying to run an intrusive check")
+	klog.Info("Trying to run an invasive check")
 	utils.HealthcheckLock.Lock()
 	defer utils.HealthcheckLock.Unlock()
 	if utils.GPUsAvailability() {
-		utils.CreateJob("dcgm")
+		klog.Info("Begining invasive health checks, updating node label =TESTING for node ", os.Getenv("NODE_NAME"))
+		label := `
+		{
+			"metadata": {
+				"labels": {
+					"autopilot.ibm.com/gpuhealth": "TESTING"
+					}
+			}
+		}
+		`
+		utils.PatchNode(label, os.Getenv("NODE_NAME"))
+		err := utils.CreateJob("dcgm")
+		if err != nil {
+			klog.Info("Invasive health checks failed, updating node label for node ", os.Getenv("NODE_NAME"))
+			label := `
+		{
+			"metadata": {
+				"labels": {
+					"autopilot.ibm.com/gpuhealth": ""
```

Comment:
@cmisale should a failed `dcgm` job results in `gpuhealth=ERR` ? Here I've cleared the label if it errrors (not sure if that's the right thing to do in any case)

Reply:
That is a little more complicated..
No, a fail in the creation of a Job for the dcgm should not result into an error in the GPUs.
In this code, I am not sure how to interpret this result. Because again, a periodic check that reports all good, is going to overwrite that label with `PASS`.  And there it's clearing out that label for no real reason... meaning the reason is not related to the GPU failings or any other test.

If we want to update that value with `PASS` or `ERR` based on the dcgm r3 result, it has to happen somewhere else.

Reply:
> If we want to update that value with PASS or ERR based on the dcgm r3 result, it has to happen somewhere else.
Yes, it should. My assumption in this PR was that TESTING label would only persist until the dcgm result finished, when it was overwritten with PASS or ERR.

Reply:
It does get overwritten here: https://github.com/IBM/autopilot/blob/b25afcacf85b140cb34584db4c2a5a42ca848005/autopilot-daemon/gpu-dcgm/entrypoint.py#L230

Reply:
Good point. So that should close the loop! We only need to make sure that the `testing` value goes away if something happens and can't remove it. This is quite critical because it would prevent jobs from running even tho the node might be perfectly fine

Reply:
If `CreateJob(...)` fails then the `TESTING` label is cleared. If `gpu-dcgm/entrypoint.py` starts it _should_ eventually overwrite `TESTING` with a `PASS` or `EVICT` label. 

I suppose `gpu-dcgm/entrypoint.py` could crash before updating the label, and then we would be in trouble. However, we don't want to remove TESTING before dcgm finishes... Hmm. any ideas?

Reply:
I don't have a solution for this.. 
I think the place where it can go wrong, is in the patch api call, we *should* be managing the dcgm fails/errors in the `try_dcgm` function